### PR TITLE
Don't assert over invalid channel args

### PR DESCRIPTION
**Pull request by dgquintas**
Fixes
- https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=590
- https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=587

### CODE REVIEW

File: src/core/ext/client_channel/client_channel.c [hunk 1]
```diff
@@ -543,7 +543,10 @@ static grpc_error *cc_init_channel_elem(grpc_exec_ctx *exec_ctx,
                                         grpc_channel_element *elem,
                                         grpc_channel_element_args *args) {
   channel_data *chand = elem->channel_data;
-  GPR_ASSERT(args->is_last);
+  memset(chand, 0, sizeof(*chand));
+  if (!args->is_last) {
```
Comment (ctiller) [inline src/core/ext/client_channel/client_channel.c]: This should stay an assert: there should be no way from API to get to the point that client_channel is not the last thing in the stack, and having it be elsewhere is going to cause bugs.

Comment (dgquintas) [inline src/core/ext/client_channel/client_channel.c]: Done. PTAL.

**Comment by dgquintas** [comment]
jenkins test this please

**Comment by dgquintas** [comment]
@ctiller what are the units in the benchmark check? I'm surprised this PR affects timings of hpackencoder at all

**Comment by ctiller** [comment]
Nanoseconds. It probably didn't: I'm still tuning the parameters - our
environment is a little noisier than I'd like!

**Comment by dgquintas** [comment]
Issues: #10434 #10435 #10243

**Pull request merged**